frontend/src/scenes/userLogic.tsx

```diff
@@ -5,6 +5,7 @@ import { userLogicType } from './userLogicType'
 import { UserType } from '~/types'
 import posthog from 'posthog-js'
 import { toast } from 'react-toastify'
+import { getAppContext } from 'lib/utils/getAppContext'
```

Comment:
With these changes, should we simplify the logic we introduced in the previous PR? i.e. instead of showing the loader after 1s, we could just show it right away if something failed with the context preload and we have to fetch the API request.

Reply:
I'd still show it after a second, and show nothing before, as this actually [feels faster](https://www.nngroup.com/articles/progress-indicators/) to the user.

> This indicator should be reserved for actions that take between 2-10 seconds. For anything that takes less than 1 second to load, it is distracting to use a looped animation, because users cannot keep up with what happened and might feel anxious about whatever flashed on the screen.